packages/ti_domaintools/data_stream/domaindiscovery/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,59 @@
+---
+description: Pipeline for processing domaindiscovery feed
+processors:
+- json:
+    if: ctx?.message != null
+    field: message
+    target_field: domaintools
+- set:
+    field: domaintools.feed
+    value: 'domaindiscovery_feed'
+
+  ############################
+  # Generic indicator fields #
+  ############################
+
+- set:
+    field: threat.indicator.type
+    value: domain-name
+- set:
+    if: ctx.domaintools != null && ctx.domaintools?.domain != null
+    field: threat.indicator.name
+    copy_from: domaintools.domain
+
+  ######################
+  # Threat feed fields #
+  ######################
+
+- set:
+    field: threat.feed.description
+    value: 'New domains as they are either discovered in domain registration information, observed by our global sensor network, or reported by trusted third parties.'
+- set:
+    field: threat.feed.name
+    value: 'DomainTools domaindiscovery'
+- set:
+    field: threat.feed.reference
+    value: https://docs.techdocs.ci.domaintools.cloud/feeds/realtime/userguide/
+
+  ####################
+  # Event ECS fields #
+  ####################
+
+- set:
+    field: ecs.version
+    value: '8.11.0'
+- set:
+    field: event.kind
+    value: enrichment
+- set:
+    field: event.category
+    value: ['threat']
+- set:
+    field: event.type
+    value: ['indicator']
+
+on_failure:
+- set:
+    field: error.message
+    value: '{{ _ingest.on_failure_message }}'
```

```suggestion
    value: '{{{ _ingest.on_failure_message }}}'
```
https://linux.die.net/man/5/mustache

> All variables are HTML escaped by default. If you want to return unescaped HTML, use the triple mustache: {{{name}}}.

But also, we would prefer if there were tags in the processors and reporting here for what it was that failed. See example [here](https://github.com/elastic/integrations/blob/08eafa32049cae83d822bda9ea1c5e2f5f87da22/packages/abnormal_security/data_stream/ai_security_mailbox/elasticsearch/ingest_pipeline/default.yml#L274-L295).

---

done